lib/gold_miner/slack/client.rb

```diff
@@ -24,81 +23,61 @@ def self.build(api_token:, slack_client: ::Slack::Web::Client, author_config: Au
       @slack = slack_client.new(token: api_token)
-      @author_config = author_config
+      @user_name_cache = {}
     end
 
     def auth_test
       @slack.auth_test
     end
 
-    def search_interesting_messages_in(channel)
-      interesting_messages = interesting_messages_query(channel)
-      Sync do
-        til_messages_task = Async do
-          extract_messages_from_result(
-            @slack.search_messages(query: interesting_messages.with_topic("TIL"))
-          )
-        end
-        tip_messages_task = Async do
-          extract_messages_from_result(
-            @slack.search_messages(query: interesting_messages.with_topic("tip"))
-          )
-        end
-        golden_messages_task = Async do
-          extract_messages_from_result(
-            @slack.search_messages(query: interesting_messages.with_reaction(GOLD_EMOJI))
-          )
-        end
-
-        [til_messages_task, tip_messages_task, golden_messages_task]
-          .flat_map(&:wait)
-          .uniq { |message| message[:permalink] }
-      end
+    def search_messages(query)
+      @slack
+        .search_messages(query:)
+        .tap { |response|
+          warn_on_multiple_pages(response)
+          fetch_author_names(response)
+        }
     end
 
     private_class_method :new
 
     private
 
-    def interesting_messages_query(channel)
-      MessagesQuery
-        .new
-        .on_channel(channel)
-        .sent_after_last_friday
+    # For simplicity, I'm not handling API pagination yet
+    def warn_on_multiple_pages(result)
+      if result.messages.paging.pages > 1
+        warn "[WARNING] Found more than one page of results, only the first page will be processed"
+      end
     end
 
-    def extract_messages_from_result(result)
-      warn_on_multiple_pages(result)
+    # Unfortunately, the Slack API doesn't return the real name of the
+    # author of a message, so we need to make an additional API call for
+    # each message.
+    def fetch_author_names(response)
+      Sync do
+        author_names = response.messages.matches.map { |message|
+          Async { [message.id, real_name_for(message.user)] }
+        }.map(&:wait).to_h
 
-      result.messages.matches.map do |match|
-        Slack::Message.new(
-          text: match.text,
-          author: build_author(match),
-          permalink: match.permalink
-        )
+        response.messages.matches.each { |message|
+          message.author_real_name = author_names[message.id]
+        }
       end
     end
 
-    def build_author(message)
-      Slack::User.new(
-        id: message.user,
-        name: fetch_author(message.user),
-        username: message.username,
-        link: @author_config.link_for(message.username)
-      )
+    def real_name_for(user_id)
+      @user_name_cache[user_id] ||=
+        user_info(user_id)
+          .user
+          .profile
+          .real_name
+          .tap { |name| @user_name_cache[user_id] = name }
```

Comment:
I didn't test it, but I think we don't need this last line:
```rb
.tap { |name| @user_name_cache[user_id] = name }`
```
because we are already assigning at line 71 with:
```rb
@user_name_cache[user_id] ||=
```